docparse/jsonschema.go

```diff
@@ -208,10 +208,23 @@ func fieldToSchema(prog *Program, fName string, ref Reference, f *ast.Field) (*S
 start:
 	switch typ := sw.(type) {
 
-	// Don't support interface{} for now. We'd have to add a lot of complexity
-	// for it, and not sure if we're ever going to need it.
+	// Interface, only useful for its description.
 	case *ast.InterfaceType:
-		return nil, errors.New("fieldToSchema: interface{} is not supported")
+		if len(f.Names) == 0 {
+			return nil, errors.New("fieldToSchema: interface{} is not supported")
+		}
+
+		field := f.Names[0].Obj.Decl.(*ast.Field)
+		switch typ := field.Type.(type) {
```

Comment:
This is pretty nasty, I don't know a better way of doing it :-(

Reply:
AFAIK, this is the way of doing it.